cuda_core/cuda/core/experimental/_linker.py

```diff
@@ -0,0 +1,170 @@
+
+
+    def __post_init__(self):
+        self.formatted_options = []
+        if self.arch is not None:
+            self.formatted_options.append(f"-arch={self.arch}")
+        if self.max_register_count is not None:
+            self.formatted_options.append(f"-maxrregcount={self.max_register_count}")
+        if self.time is not None:
+            self.formatted_options.append("-time")
+        if self.verbose is not None:
+            self.formatted_options.append("-verbose")
+        if self.link_time_optimization is not None:
+            self.formatted_options.append("-lto")
+        if self.ptx is not None:
+            self.formatted_options.append("-ptx")
+        if self.optimization_level is not None:
+            self.formatted_options.append(f"-O{self.optimization_level}")
+        if self.debug is not None:
+            self.formatted_options.append("-g")
+        if self.lineinfo is not None:
+            self.formatted_options.append("-lineinfo")
+        if self.ftz is not None:
+            self.formatted_options.append(f"-ftz={'true' if self.ftz else 'false'}")
+        if self.prec_div is not None:
+            self.formatted_options.append(f"-prec-div={'true' if self.prec_div else 'false'}")
+        if self.prec_sqrt is not None:
+            self.formatted_options.append(f"-prec-sqrt={'true' if self.prec_sqrt else 'false'}")
+        if self.fma is not None:
+            self.formatted_options.append(f"-fma={'true' if self.fma else 'false'}")
+        if self.kernels_used is not None:
+            for kernel in self.kernels_used:
+                self.formatted_options.append(f"-kernels-used={kernel}")
+        if self.variables_used is not None:
+            for variable in self.variables_used:
+                self.formatted_options.append(f"-variables-used={variable}")
+        if self.optimize_unused_variables is not None:
+            self.formatted_options.append("-optimize-unused-variables")
+        if self.xptxas is not None:
+            for opt in self.xptxas:
+                self.formatted_options.append(f"-Xptxas={opt}")
+        if self.split_compile is not None:
+            self.formatted_options.append(f"-split-compile={self.split_compile}")
+        if self.split_compile_extended is not None:
+            self.formatted_options.append(f"-split-compile-extended={self.split_compile_extended}")
+        if self.jump_table_density is not None:
+            self.formatted_options.append(f"-jump-table-density={self.jump_table_density}")
+        if self.no_cache is not None:
+            self.formatted_options.append("-no-cache")
+        if self.device_stack_protector is not None:
+            self.formatted_options.append("-device-stack-protector")
+
+
+class Linker:
+
+    __slots__ = ("_handle")
+
+    def __init__(self, options: LinkerOptions, object_codes = None):

```

Comment:
I think we need to revert the order:
```suggestion
    def __init__(self, *object_codes, options: LinkerOptions = None):
```
Taking `*object_codes` also allows us to loop over it later.